### PR TITLE
feat: Added Two channel mode in Logic Analyzer

### LINKED ISSUE #1124: Two Channel Mode not working in Logic Analyzer
**Actual Behaviour**

Two Channel Mode isn't working in Logic Analyzer Instrument

**Expected Behaviour**

It should work.

**Steps to reproduce it**

Go to Logic Analyzer Instrument -> Select Two Channel Mode -> Analyze

**LogCat for the issue**

Not applicable

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by harsh-2711**
Fixes #1124 

**Changes**: Added Two Channel Mode for Logic Analyzer including all edges support

**Screenshot/s for the changes**: 
screenshot_20180627-140611
screenshot_20180627-140657
screenshot_20180627-140717

Please use Control Activity to generate waves for testing the apk as Wave Generator is still not generating digital waves for all pins

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**:  
[app-debug.zip](https://github.com/fossasia/pslab-android/files/2140444/app-debug.zip)

NOTE: Please merge PR #1137 before merging this PR

### CODE REVIEW

**Comment by harsh-2711** [comment]
@CloudyPadmal Please add WIP tag.

**Comment by harsh-2711** [comment]
@abhinavraj23 The changes done in PR #1137 was needed in this PR. So, I made a branch of that PR. It's not a good practice to combine them. Also, I have added a note for the same in the main comment of this PR. Take a look

**Comment by harsh-2711** [comment]
@CloudyPadmal Now need to add WIP tag. I have completed the implementation and updated the apk and screenshots. @CloudyPadmal @Avjeet @abhinavraj23 @Vikum94 Please review

**Comment by CloudyPadmal** [comment]
@abhinavraj23 @Avjeet @Vikum94 please run this APK, test functionalities and confirm it's functional. Let's speed this up! :+1:

File: app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java [hunk 1]
```diff
@@ -477,8 +531,19 @@ private void singleChannelOtherEdges(double[] xData, double[] yData) {
         for (int i = 0; i < xData.length; i++) {
             int xaxis = (int) xData[i];
             int yaxis = (int) yData[i];
-            tempInput.add(new Entry(xaxis, yaxis));
+            tempInput.add(new Entry(xaxis, yaxis + 2 * currentChannel));
         }
+
+        // Adding the entry for current channel
+        LineDataSet lineDataSet = new LineDataSet(tempInput, channelNames.get(currentChannel));
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java]: @harsh-2711 I observed these set of lines in three places. Please create a method with parameters like linedataset and call that method in these three places.

File: app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java [hunk 2]
```diff
@@ -400,74 +435,93 @@ public void onNothingSelected() {
...
+                    tempInput.add(new Entry(xaxis, 0 + 2 * currentChannel));
                     count = 0;
                 } else
                     count++;
                 check = xaxis;
             }
         }
+
+        // Adding the entry for current channel
+        LineDataSet lineDataSet = new LineDataSet(tempInput, channelNames.get(currentChannel));
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java]: Similar code, implement using a method.

File: app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java [hunk 3]
```diff
@@ -400,74 +435,93 @@ public void onNothingSelected() {
...
-                    tempInput.add(new Entry(xaxis.get(i), yaxis.get(i + 1)));
-                }
 
-            }
-            tempInput.add(new Entry(xaxis.get(xaxis.size() - 1), yaxis.get(xaxis.size() - 1)));
         }
+        tempInput.add(new Entry(xaxis.get(xaxis.size() - 1), yaxis.get(xaxis.size() - 1) + 2 * currentChannel));
+
+        // Adding the entry for current channel
+        LineDataSet lineDataSet = new LineDataSet(tempInput, channelNames.get(currentChannel));
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java]: Same here

**Comment by harsh-2711** [comment]
@CloudyPadmal Changes done.

File: app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java [hunk 4]
```diff
@@ -2007,28 +2007,24 @@ public boolean fetchLAChannels() {
...
 
         LinkedHashMap<String, Integer> tempMap = new LinkedHashMap<>();
-        tempMap.put("ID1", initialStates.get("ID1"));
-        tempMap.put("ID2", initialStates.get("ID2"));
-        tempMap.put("ID3", initialStates.get("ID3"));
-        tempMap.put("ID4", initialStates.get("ID4"));
+        tempMap.put("ID1", initialStates.get("A"));
+        tempMap.put("ID2", initialStates.get("B"));
+        tempMap.put("ID3", initialStates.get("C"));
+        tempMap.put("ID4", initialStates.get("D"));
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java]: Can't we use `IDx` instead of ABCD here?

**Comment by harsh-2711** [comment]
@CloudyPadmal Changes done. Sorry for repeating the notation.

**Pull request merged**